CMakeLists.txt

```diff
@@ -35,6 +35,7 @@ option(AVM_RELEASE "Build an AtomVM release" OFF)
 option(AVM_CREATE_STACKTRACES "Create stacktraces" ON)
 option(AVM_BUILD_RUNTIME_ONLY "Only build the AtomVM runtime" OFF)
 option(COVERAGE "Build for code coverage" OFF)
+option(ENABLE_ASAN "Use Address Sanitizer" ON)
```

Comment:
I'm not sure if this should be the default: it is very useful during development, but it should be off during release.
Also I'm not sure how interacts with platforms that do not properly support `-fsanitize=address` such as ESP32.

So maybe it should be manually enabled (and we might suggest it in our documentation) and of course we should add a job in our CI that enables it. After all what really matter is not merging code that fails those checks.

I think we should have jobs and options for undefined behavior sanitizer, thread sanitizer and memory sanitizer and they should be part of our CI as well.

Reply:
Makes sense. I don't have means to test embedded platforms so most of my testing comes from unix and WASM port.

What do you think about adding
```
set(SANITIZER none) # available values: address, memory, undefined, none
```

and corresponding ifs in libAtomVM CMakeList?

Is there any way to create local CMakeLists or you need to use env variables when running CMake to set those options?